packages/plugin-hardhat/src/defender/deploy.ts

```diff
@@ -103,7 +105,7 @@ export async function defenderDeploy(
     network: network,
     artifactPayload: JSON.stringify(contractInfo.buildInfo),
     licenseType: licenseType,
-    constructorInputs: constructorArgs,
+    constructorInputs: args as (string | number | boolean)[],
```

Comment:
It looks like this that doesn't support constructor argument that are arrays or structs

For example
```solidity
constructor(address[] memory admins) {
    for (uint256 i = 0; i < admins.length; ++i) _grantRole(DEFAULT_ADMIN_ROLE, admins[i]);
}
```

Imo this should be 
```suggestion
    constructorInputs: args as any[],
```

Reply:
Defender SDK requires the former type, but this does look more correct and I tested that your constructor example works through Defender.

Reply:
Hi, just adding context on this, I remember we had some conversations with @ericglau about how to handle higher order types as constructor args in defender deployments.
The main issue was that there was no easy way to serialize arrays or structs objects, so we decided to support `constructorBytecode` in our API for those cases, meaning that in cases of arrays or higher order types, we delegate the conversion to bytecode in the client side
Here an example of that:
https://github.com/OpenZeppelin/defender-sdk/blob/main/examples/deploy-contract/index.js#L37

Reply:
@MCarlomagno Thanks for the context. Based on this, we should switch the Hardhat plugin to encode the calldata first, and pass it through using `constructorBytecode` to avoid this limitation.  We also use `constructorBytecode` [in Foundry](https://github.com/OpenZeppelin/openzeppelin-foundry-upgrades/blob/main/src/internal/DefenderDeploy.sol#L84).

Reply:
Yes! Correct